mono/metadata/handle.h

```diff
@@ -245,7 +245,7 @@ Icall macros
 	do {							\
 		void* __result = MONO_HANDLE_RAW (HANDLE);	\
 		CLEAR_ICALL_FRAME;				\
-		return __result;				\
+		return g_cast (__result);			\
```

Comment:
@luhenry this use of `g_cast` in `HANDLE_FUNCTION_RETURN_OBJ` is difficult to avoid unless we change the macro to take a return type.

Instead of this:

```c
MONO_RT_EXTERNAL_ONLY
MonoString *
some_api_function (const char *s)
{
  HANDLE_FUNCTION_ENTER();
  MonoStringHandle h = mono_string_new_handle (s);
  HANDLE_FUNCTION_RETURN_OBJ (h);
}
```

We would have to write

```c
MonoString *
some_api_function (const char *s)
{
  HANDLE_FUNCTION_ENTER ();
  MonoStringHandle h = mono_string_new_handle (s);
  HANDLE_FUNCTION_RETURN_OBJ (MonoString, h);
}
```

Reply:
This one can use `auto` instead. Two of the handle.h changes can be `auto`, the other one I didn't look at closely. Because the type usually matches from `MONO_HANDLE_RAW`.